backend/app/routes.py

```diff
@@ -136,8 +136,39 @@ def get_links_between_two_nodes(from_node_id, to_node_id):
         abort(400, description="Source node can not be the same as target node.")
         return
 
-    shortest_link_query_result = db.session.query(func.get_links_btwn_nodes(from_node_id, to_node_id)).first()[0]
-    shortest_link_data = parse_get_links_btwn_nodes_response(shortest_link_query_result)
+    connection = getConnection()
+
+    with connection:
+        with connection.cursor() as cursor:
+            #Uses pg_routing to route between the start node and end node on the HERE
+            #links network. Returns inputs and an array of link_dirs and a unioned line
+            select_sql = '''
+                WITH results as (
+                    SELECT *
+                    FROM pgr_dijkstra(
+                        'SELECT id, source::int, target::int, length::int AS cost FROM here.routing_streets_name',
+                        %(node_start)s,
+                        %(node_end)s
+                    )
+                )
+
+                SELECT 
+                    %(node_start)s,
+                    %(node_end)s,
+                    array_agg(st_name),
+                    array_agg(link_dir),
+                    ST_AsGeoJSON(ST_union(ST_linemerge(geom))) AS geometry
+                FROM results
+                INNER JOIN here.routing_streets_name on edge = id'''
+            cursor.execute(select_sql, {"node_start": from_node_id, "node_end": to_node_id})
+            ( source, target, path, link_dirs, geometry ) = cursor.fetchall()[0]
+            print(type(geometry))
+
+    shortest_link_data = {"source": source, "target": target,
+        "path_name": str(path), "link_dirs": link_dirs, "geometry": json.loads(geometry)}

```

Comment:
`path` is an array of strings. `str(path)` converts the whole array to one big string.

Reply:
I just inserted the whole list as a placeholder, not sure if we want to create a naming system for routes

Reply:
What was it doing before? The one thing I noticed was that this feeds into the names that appear when you hover over a segment on the map.

Reply:
New code: removes duplicates from list. The new path_name is the casted string of the list of unique stnames.

Reply:
Thinking about it, this is pretty related to issue #15 yes?